pkg/reconciler/v1alpha1/namespace/namespace.go

```diff
@@ -18,26 +18,114 @@ package namespace
 
 import (
 	"context"
-
 	"github.com/knative/eventing/contrib/gcppubsub/pkg/util/logging"
 	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
 	"go.uber.org/zap"
+	"k8s.io/api/core/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
 const (
+	// controllerAgentName is the string used by this controller to identify
+	// itself when creating events.
+	controllerAgentName = "knative-eventing-namespace-controller"
+
 	defaultBroker             = "default"
 	knativeEventingAnnotation = "eventing.knative.dev/injection"
 
 	// Name of the corev1.Events emitted from the reconciliation process
 	brokerCreated = "BrokerCreated"
 )
 
+type reconciler struct {
+	client        client.Client
+	restConfig    *rest.Config
+	dynamicClient dynamic.Interface
+	recorder      record.EventRecorder
+
+	logger *zap.Logger
+}
+
+// Verify the struct implements reconcile.Reconciler
+var _ reconcile.Reconciler = &reconciler{}
+
+// ProvideController returns a function that returns a Broker controller.
+func ProvideController(logger *zap.Logger) func(manager.Manager) (controller.Controller, error) {
+	return func(mgr manager.Manager) (controller.Controller, error) {
+		// Setup a new controller to Reconcile Brokers.
+		r := &reconciler{
+			recorder: mgr.GetRecorder(controllerAgentName),
+			logger:   logger,
+		}
+		c, err := controller.New(controllerAgentName, mgr, controller.Options{
+			Reconciler: r,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		// Watch Namespaces.
+		if err = c.Watch(&source.Kind{Type: &v1.Namespace{}}, &handler.EnqueueRequestForObject{}); err != nil {
+			return nil, err
+		}
+
+		// Watch all the resources that this reconciler reconciles.
+		if err = c.Watch(&source.Kind{Type: &v1alpha1.Broker{}}, &handler.EnqueueRequestsFromMapFunc{ToRequests: &namespaceMapper{}}); err != nil {
+			return nil, err
+		}
+
+		for _, t := range []runtime.Object{ &v1alpha1.Broker{} } {
```

Comment:
this for loop seems redundant. The previous line adds the broker. I may be missing something. Please double check

Reply:
Removed the previous one.